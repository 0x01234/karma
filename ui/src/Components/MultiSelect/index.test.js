import React from "react";

import { shallow, mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { MultiSelect } from ".";

const Option = value => ({ label: value, value: value });

describe("<MultiSelect />", () => {
  it("renders without any extra props", () => {
    const tree = shallow(<MultiSelect />);
    expect(tree.text()).toBe("<StateManager />");
  });
});

class CustomMultiSelect extends MultiSelect {
  constructor(props) {
    super(props);
    this.extraProps = props;
  }

  renderProps = () => this.extraProps;
}

describe("<CustomMultiSelect />", () => {
  it("matches snapshot with defaults", () => {
    const tree = shallow(<CustomMultiSelect />);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with isMulti=true", () => {
    const tree = shallow(<CustomMultiSelect isMulti />);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot when focused", () => {
    // this test is to cover styles state.isFocused conditions
    const tree = mount(<CustomMultiSelect autoFocus />);
    tree.find("input").simulate("focus");
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with a value", () => {
    const tree = shallow(
      <CustomMultiSelect
        defaultValue={Option("foo")}
        options={[Option("foo", Option("bar"))]}
      />
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with isMulti=true and a value", () => {
    const tree = shallow(
      <CustomMultiSelect
        isMulti
        defaultValue={Option("foo")}
        options={[Option("foo", Option("bar"))]}
      />
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });
});
