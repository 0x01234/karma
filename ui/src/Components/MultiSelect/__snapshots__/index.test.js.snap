// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomMultiSelect /> matches snapshot with a value 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={
    Object {
      "label": "foo",
      "value": "foo",
    }
  }
  options={
    Array [
      Object {
        "label": "foo",
        "value": "foo",
      },
    ]
  }
  styles={
    Object {
      "control": [Function],
      "indicatorsContainer": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "multiValueRemove": [Function],
      "option": [Function],
      "valueContainer": [Function],
    }
  }
/>
`;

exports[`<CustomMultiSelect /> matches snapshot with defaults 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  styles={
    Object {
      "control": [Function],
      "indicatorsContainer": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "multiValueRemove": [Function],
      "option": [Function],
      "valueContainer": [Function],
    }
  }
/>
`;

exports[`<CustomMultiSelect /> matches snapshot with isMulti=true 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isMulti={true}
  styles={
    Object {
      "control": [Function],
      "indicatorsContainer": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "multiValueRemove": [Function],
      "option": [Function],
      "valueContainer": [Function],
    }
  }
/>
`;

exports[`<CustomMultiSelect /> matches snapshot with isMulti=true and a value 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={
    Object {
      "label": "foo",
      "value": "foo",
    }
  }
  isMulti={true}
  options={
    Array [
      Object {
        "label": "foo",
        "value": "foo",
      },
    ]
  }
  styles={
    Object {
      "control": [Function],
      "indicatorsContainer": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "multiValueRemove": [Function],
      "option": [Function],
      "valueContainer": [Function],
    }
  }
/>
`;
